src/openms/source/ANALYSIS/OPENSWATH/TransitionPQPFile.cpp

```diff
@@ -456,10 +489,18 @@ namespace OpenMS
       int group_set_index = group_map[peptide.id];
       int peptide_set_index = peptide_map[peptide_sequence];
 
-      for (std::vector<String>::iterator it = peptide.protein_refs.begin(); it != peptide.protein_refs.end(); ++it)
+      for (const auto& it : peptide.protein_refs)
+      {
+        peptide_protein_map.push_back(std::make_pair(peptide_set_index, protein_map[it]));
```

Comment:
performance opt: `peptide_protein_map.emplace_back(peptide_set_index, protein_map[it]);`